tests/platform_tests/test_memory_exhaustion.py

```diff
@@ -41,6 +42,12 @@ def tearDown(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, pdu_co
                           'Recover {} by PDU reboot failed'.format(hostname))
             # Wait until all critical processes are healthy.
             wait_critical_processes(duthost)
+            # For sup, we also need to ensure linecards are back and healthy for following tests
+            is_sup = duthost.get_facts().get("modular_chassis") and duthost.is_supervisor_node()
+            if is_sup:
+                for lc in duthosts.frontend_nodes:
+                    wait_for_startup(lc, localhost, delay=10, timeout=300)
+                    wait_critical_processes(lc)
```

Comment:
can we write commmon function for this code, it is being used couple time here may be useful in other tests as well